index.d.ts

```diff
@@ -1156,7 +1156,11 @@ declare namespace Eris {
   export class Collection<T extends { id: string | number }> extends Map<string | number, T> {
     baseObject: new (...args: any[]) => T;
     limit?: number;
-    constructor(baseObject: new (...args: any[]) => T, limit?: number);
+    constructor(baseObject: new (...args: any[]) => T, limit?: number, iterable?: object | any[]);
+    static from(array: any[], key: string, limit: number): Collection<any>;
+    toArray(): any[];
+    toObject(): object;
```

Comment:
I don't like this. Theoretically, this is useless to typings. All you will get is `{}` and TS won't let you use it at all since `anyprop` does not exist on `{}`. I would prefer `unknown` to be more typesafe.
```suggestion
    toObject(): unknown;
```

Reply:
Why would you make this `unknown`? That conveys even less than `object` does (`object` matches any non-primitive, not just `{}`.)

A better type, since you've proposed added generics to the existing typings, would be:
```suggestion
    toObject(): {[key: T["id"]]: T};
```

Reply:
Same here, doesn't seem to be able to do what Geo suggests.
Complaining about the `key` part;
Maybe I should do it anyway, not really familiar with ts / typings

Reply:
If my suggestion above causes issues, using the `object` type is probably a safe bet. I'll look into it in a bit.

Reply:
I disagree about using `object`. Object is too broad and can cause incorrect typings issues. It is not typesafe. 

```js
const test = (idk: object) => {
    console.log(idk)
  }

  // Valid
  test(test)
  // Valid
  test({ idk: 'test' })
```

Having it return `object` can break a lot of typings which lead to potential bugs/errors. `object` will make it so it will be treated as an array, function, anything really. object is a useless and bad type to be using.

The worst part is if you tried to do object.id it will throw typings errors.
```js
const test = (idk: object) => {
    console.log(idk.id)
  }
```

Using `unknown` is a much better alternative. It is typesafe.

If possible id actually prefer using ur suggested change. I think that is better than `unknown` but if its a choice between object or unknown, `unknown` is the better choice.

Reply:
Most of the issues with using `object` apply when using it as the type of a parameter, but disappear when using it as the type of a return value. It just requires the consumer to do more type checks at runtime since TS doesn't have the type info.

Regardless, I've verified a fixed version of my suggestion above. The way TS handles index signatures is weird, so what we really want is this:

```suggestion
    toObject(): {[key: string]: T | undefined};
```

Using a `number` as a key works fine even when the index signature is of type `string` since all object keys are strings (or symbols, but we don't care about those) internally. I also forgot to use `| undefined` before.